src/storage/file_io.c

```diff
@@ -7163,28 +7161,15 @@ fileio_start_backup (THREAD_ENTRY * thread_p, const char *db_full_name_p, INT64
       backup_header_p->bkpagesize *= FILEIO_FULL_LEVEL_EXP;
     }
 
-  switch (zip_method)
+  backup_header_p->zip_method = zip_method;
+  backup_header_p->zip_level = zip_level;
+
+  if (backup_header_p->zip_method == FILEIO_ZIP_LZ4_METHOD
+      && FILEIO_DBVOLS_IO_PAGE_SIZE (backup_header_p) > LZ4_MAX_INPUT_SIZE)
```

Comment:
I think that backup file size limitation seems to compromise backward compatibility.

Reply:
From checking though `file_io.c`, there is a limit on the size of the buffers that are compressed/decompressed during backup/restore. `fileio_restore_volume's` decompresses nbytes. By tracking nbytes related stuff, we can find `bkpagesize ~ bkuphdr->bkpagesize <= (bkuphdr->db_iopagesize * 32)` and `db_iopagesize` seems to be upper-bounded by `IO_MAX_PAGE_SIZE` (16K). So, file_io.c seems to compress/decompress at most around 32 * 16 * 1024 bytes, making it not an issue of backward compatibility.
Although, it might hinder potential future usage due to the LZ4_MAX_INPUT_SIZE value, compressing bigger stuff will require splitting & compressing.

Reply:
Ok, I understood.